internal/serverinstall/nomad.go

```diff
@@ -477,42 +504,7 @@ func (i *NomadInstaller) Upgrade(
 	// than the direct static IP for the CLI context and server config. Otherwise
 	// if Nomad restarts the server allocation, a new IP will be assigned and any
 	// configured clients will be invalid
-	if i.config.consulService {
-		s.Update("Configuring the server context to use Consul DNS hostname")
-		if i.config.consulDatacenter == "" {
-			i.config.consulDatacenter = defaultConsulDatacenter
-		}
-		if i.config.consulDomain == "" {
-			i.config.consulDomain = defaultConsulDomain
-		}
-
-		grpcPort, _ := strconv.Atoi(defaultGrpcPort)
-		httpPort, _ := strconv.Atoi(defaultHttpPort)
-
-		if i.config.consulServiceHostname == "" {
-			addr.Addr = fmt.Sprintf("%s.service.%s.%s:%d",
-				waypointConsulBackendName, i.config.consulDatacenter, i.config.consulDomain, grpcPort)
-			httpAddr = fmt.Sprintf("%s.service.%s.%s:%d",
-				waypointConsulUIName, i.config.consulDatacenter, i.config.consulDomain, httpPort)
-		} else {
-			addr.Addr = fmt.Sprintf("%s:%d", i.config.consulServiceHostname, grpcPort)
-			httpAddr = fmt.Sprintf("%s:%d", i.config.consulServiceHostname, httpPort)
-		}
-	} else {
-		s.Update("Configuring the server context to use the static IP address from the Nomad allocation")
-
-		serverAddr, err := getAddrFromAllocID(allocID, client)
-		if err != nil {
-			return nil, err
-		}
-		hAddr, err := getHTTPFromAllocID(allocID, client)
-		if err != nil {
-			return nil, err
-		}
-
-		httpAddr = hAddr
-		addr.Addr = serverAddr
-	}
+	httpAddr, addr.Addr, err = i.getWaypointAddress(client, allocID)
```

Comment:
We are not checking this `err`, we should, or use `_` if it's something we can ignore